### PR TITLE
Update rasa run & rasa shell

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- `rasa run` always starts Rasa Server
- Remove subparsers `rasa run nlu` and `rasa run core` as they do not differ from `rasa run`
- `rasa shell` offers possibility to chat via command line if Core model is present. Otherwise, the NLU model is used to interpret text.
- `rasa shell nlu` interpret text on the command line as long as the provided model contains an NLU model

related to https://github.com/RasaHQ/rasa/issues/3272

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/run.py [hunk 1]
```diff
@@ -47,10 +47,10 @@ def run(
...
 
     if not connector and not credentials:
-        channel = "cmdline"
-        logger.info(
+        channel = "rest"
+        print_warning(
             "No chat connector configured, falling back to the "
-            "command line. Use `rasa configure channel` to connect"
+            "rest. Use `rasa configure channel` to connect"
             "the bot to e.g. facebook messenger."
```
Comment (MetcalfeTom) [inline rasa/run.py]:
```suggestion
```

File: rasa/run.py [hunk 2]
```diff
@@ -47,10 +47,10 @@ def run(
     _endpoints = AvailableEndpoints.read_endpoints(endpoints)
 
     if not connector and not credentials:
-        channel = "cmdline"
-        logger.info(
+        channel = "rest"
+        print_warning(
             "No chat connector configured, falling back to the "
-            "command line. Use `rasa configure channel` to connect"
+            "rest. Use `rasa configure channel` to connect"
```
Comment (MetcalfeTom) [inline rasa/run.py]:
```suggestion
            "REST input channel.  To connect your bot to another channel, "
            "read the docs here: https://rasa.com/docs/core/connectors"
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -37,6 +37,7 @@ in ``interactive learning``
 - support for spaCy 2.1
 - a model for an agent can now also be loaded from a remote storage
 - log level can be set via environment variable ``LOG_LEVEL``
+- add subparser ``rasa shell nlu`` to interpret text on the command line
```
Comment (tmbo) [inline CHANGELOG.rst]: not as a change as the cli is new
```suggestion
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -66,6 +67,7 @@ Removed
 - removed ``_sklearn_numpy_warning_fix`` from the ``SklearnIntentClassifier``
 - removed ``Dispatcher`` class from core
 - removed projects: the Rasa NLU server now has a maximum of one model at a time loaded.
+- removed subparsers ``rasa run nlu`` and ``rasa run core`` as they offer the same functionality as ``rasa run``.
```
Comment (tmbo) [inline CHANGELOG.rst]:
```suggestion
```

**Pull request merged**